lib/domain-shared/shared_use_cases/get_available_balance_use_case.dart

```diff
@@ -1,11 +1,12 @@
 import 'package:async/async.dart';
 import 'package:seeds/datasource/local/settings_storage.dart';
 import 'package:seeds/datasource/remote/api/balance_repository.dart';
+import 'package:seeds/datasource/remote/model/token_model.dart';
 
 class GetAvailableBalanceUseCase {
   final BalanceRepository _balanceRepository = BalanceRepository();
 
-  Future<Result> run() {
-    return _balanceRepository.getBalance(settingsStorage.accountName);
+  Future<Result> run(TokenModel token) {
+    return _balanceRepository.getTokenBalance(settingsStorage.accountName, token: token);
```

Comment:
in here you would pass in the information from the token you need, not the entire token

Reply:
Done